src/rock-paper-scissors/services/game-process/shapes/rock.ts

```diff
@@ -0,0 +1,9 @@
+import { ShapeAbstract } from './shape-abstract';
+import { PlayerChoice, Shapes } from '../../../types/game';
+
+export class Rock extends ShapeAbstract {
+  achillesHeel: Shapes = Shapes.PAPER;
```

Comment:
Is this really needed?

Reply:
Heh,  https://en.wikipedia.org/wiki/Achilles%27_heel
That is a metaphor to shape weaknesses, each shape has it, Rock has Paper as a weakness, and so on

Reply:
I will rename in to **weakness**